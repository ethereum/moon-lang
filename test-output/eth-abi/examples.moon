abi = zb2rhbCf7LRDYaGXNxJ6vGRNXjnPaxWm7dVS6RWeVoszDkbxK
big = zb2rhnKgZ66iwb9AGyxTMP6zPbkwxe5jKxrKYTofJqisdhfJU
fromAscii = zb2rhoav2oVBSU3zQ3Yu1yZqUPcDFwLzxw2bG8NZqKWf4RVUj

{
  a:
    (abi "decode" "(bytes32[],bytes32[])"
      (abi "encode" "(bytes32[],bytes32[])"
        [["0x1234","0x1111"],["0xabcd","0x777777"]]))
  b:
    (abi "encodeCall" "sam(bytes,bool,uint256[])"
      [(fromAscii "dave") (big "1") [(big "1") (big "2") (big "3")]])
}
