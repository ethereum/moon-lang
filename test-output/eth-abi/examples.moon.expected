{
  a: [
    [
      "0x1234000000000000000000000000000000000000000000000000000000000000"
      "0x1111000000000000000000000000000000000000000000000000000000000000"
    ]
    [
      "0xabcd000000000000000000000000000000000000000000000000000000000000"
      "0x7777770000000000000000000000000000000000000000000000000000000000"
    ]
  ]
  b: "0xa5643bf20000000000000000000000000000000000000000000000000000000000000060000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000000464617665000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000003000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000003"
}
